projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/options/LLVMBaseOption.java

```diff
@@ -29,131 +29,61 @@
-    OPTIMIZATION_VALUE_PROFILE_FUNCTION_ARGS("ValueProfileFunctionArgs", "Enable value profiling for function arguments", true, LLVMOptions::parseBoolean, PropertyCategory.PERFORMANCE),
-    OPTIMIZATION_BRANCH_PROBABILITIES("InjectBranchProbabilities", "Injects branch probabilities for the basic block successors", true, LLVMOptions::parseBoolean, PropertyCategory.PERFORMANCE),
-    OPTIMIZATION_INTRINSIFY_C_FUNCTIONS("IntrinsifyCFunctions", "Substitute C functions by Java equivalents where possible", true, LLVMOptions::parseBoolean, PropertyCategory.PERFORMANCE),
-    OPTIMIZATION_INLINE_CACHE_SIZE("InlineCacheSize", "Specifies the size of the polymorphic inline cache", 5, LLVMOptions::parseInteger, PropertyCategory.PERFORMANCE),
-    OPTIMIZATION_LIFE_TIME_ANALYSIS(
-                    "EnableLifetimeAnalysis",
-                    "Performs a lifetime analysis to set dead frame slots to null to assist the PE",
-                    true,
-                    LLVMOptions::parseBoolean,
-                    PropertyCategory.PERFORMANCE),
-    NATIVE_CALL_STATS("PrintNativeCallStats", "Outputs stats about native call site frequencies", false, LLVMOptions::parseBoolean, PropertyCategory.DEBUG),
-    LIFE_TIME_ANALYSIS_STATS("PrintLifetimeAnalysisStats", "Outputs the results of the lifetime analysis (if enabled)", false, LLVMOptions::parseBoolean, PropertyCategory.DEBUG),
-    TRACE_EXECUTION("TraceExecution", "Trace execution, printing each SSA assignment", false, LLVMOptions::parseBoolean, PropertyCategory.DEBUG);
-
-    LLVMBaseOption(String key, String description, Object defaultValue, OptionParser parser, PropertyCategory category) {
-        this.key = LLVMOptions.getOptionPrefix() + key;
-        this.description = description;
-        this.defaultValue = defaultValue;
-        this.parser = parser;
-        this.category = category;
-    }
-
-    private final String key;
-    private final String description;
-    private final Object defaultValue;
-    private final OptionParser parser;
-    private final PropertyCategory category;
-
-    @Override
-    public String getKey() {
-        return key;
-    }
-
-    @Override
-    public String getDescription() {
-        return description;
-    }
-
-    @Override
-    public Object getDefaultValue() {
-        return defaultValue;
-    }
-
-    @Override
-    public String getCategoryLabel() {
-        return category.toString();
-    }
-
-    @Override
-    public Object getValue() {
-        return parser.parse(this);
-    }
-
-    public enum PropertyCategory {
-        GENERAL,
-        DEBUG,
-        PERFORMANCE,
-        TESTS,
-        MX;
-    }
+import com.oracle.truffle.llvm.option.Constants;
+import com.oracle.truffle.llvm.option.Option;
+import com.oracle.truffle.llvm.option.OptionCategory;
+
+@OptionCategory(name = "Base Options")
```

Comment:
Could we put the debug options into a different category, similar as in the previous implementation? I think reading through a list of 16 options is difficult to read.